src/ingestor/writer/settings.ts

```diff
@@ -19,14 +19,26 @@ export async function verify(body): Promise<any> {
   }
 
   const spaceUri = await snapshot.utils.getSpaceUri(msg.space, DEFAULT_NETWORK);
-  const isOwner = spaceUri.includes(body.address);
+
+  let isOwner = false;
+  let isSameSpace = false;
+  const spaceUriArray = spaceUri?.split('/') ?? [];
+  if (DEFAULT_NETWORK === '1') {
+    isOwner = spaceUriArray[4] === body.address;
+    isSameSpace = spaceUriArray[5] === msg.space;
+  } else if (spaceUriArray[4] === 'testnet') {
+    isOwner = spaceUriArray[5] === body.address;
+    isSameSpace = spaceUriArray[6] === msg.space;
+  }
```

Comment:
How about doing that way? More simple and strict verification
```js
const isOwner = spaceUri === `ipns://storage.snapshot.page/registry/${path}${body.address}/${msg.space}`
```

Reply:
@bonustrack Thanks, Changed it